rotary/rules/polish.smk

```diff
@@ -0,0 +1,360 @@
+            "{sample}/benchmarks/polish/calculate_short_read_coverage.txt"
+        threads:
+            config.get("threads",1)
+        resources:
+            mem=int(config.get("memory") / config.get("threads",1))
+        shell:
+            """
+            # Note that -F 4 removes unmapped reads
+            bwa index {input.contigs} 2> {log}
+            bwa mem -t {threads} {input.contigs} {input.qc_short_r1} {input.qc_short_r2} 2>> {log} | \
+              samtools view -b -F 4 -@ {threads} 2>> {log} | \
+              samtools sort -@ {threads} -m {resources.mem}G 2>> {log} \
+              > {output.mapping}
+            samtools index -@ {threads} {output.mapping}
+            samtools coverage {output.mapping} > {output.coverage}
+            """
+
+if (config.get("meandepth_cutoff_long_read") != "None") | (config.get("evenness_cutoff_long_read") != "None"):
+    filtration_method.append("long_read")
+
+    rule calculate_long_read_coverage:
+        input:
+            contigs="{sample}/polish/cov_filter/{sample}_pre_filtered.fasta",
+            qc_long_reads="{sample}/qc/{sample}_qc_long.fastq.gz"
+        output:
+            mapping=temp("{sample}/polish/cov_filter/{sample}_long_read.bam"),
+            mapping_index=temp("{sample}/polish/cov_filter/{sample}_long_read.bam.bai"),
+            coverage="{sample}/polish/cov_filter/{sample}_long_read_coverage.tsv",
+            long_read_map=temp("{sample}/assembly/end_repair/{sample}_repaired.fasta.map-ont.mmi")
+        conda:
+            "../envs/mapping.yaml"
+        log:
+            "{sample}/logs/polish/calculate_long_read_coverage.log"
+        benchmark:
+            "{sample}/benchmarks/polish/calculate_long_read_coverage.txt"
+        threads:
+            config.get("threads",1)
+        resources:
+            mem=int(config.get("memory") / config.get("threads",1))
+        shell:
+            """
+            # Note that -F 4 removes unmapped reads
+            minimap2 -t {threads} -ax map-ont {input.contigs} {input.qc_long_reads} 2> {log} | \
+              samtools view -b -F 4 -@ {threads} 2>> {log} | \
+              samtools sort -@ {threads} -m {resources.mem}G 2>> {log} \
+              > {output.mapping}
+            samtools index -@ {threads} {output.mapping}
+            samtools coverage {output.mapping} > {output.coverage}
+            """
+
+
+rule summarize_contigs_by_coverage:
+    input:
+        expand("{{sample}}/polish/cov_filter/{{sample}}_{type}_coverage.tsv",
+            type=filtration_method)
+    output:
+        "{sample}/polish/cov_filter/{sample}_filtered_contigs.list"
+    params:
+        meandepth_short=config.get("meandepth_cutoff_short_read"),
+        evenness_short=config.get("evenness_cutoff_short_read"),
+        meandepth_long=config.get("meandepth_cutoff_long_read"),
+        evenness_long=config.get("evenness_cutoff_long_read")
+    run:
+        #
+        def filter_coverage_data(coverage_file, mean_depth, evenness):
```

Comment:
@jmtsuji, do you have to move imports inside the run block as if it's a script? I suspect everything outside the run block can't be seen inside unless it is a special variable created by Snakemake. That is my suspicion but I am not sure, it could be injecting it differently than I expect. If we wanted to move it, I would move it `utils.py` (most of utils.py would be moved to its own library but we can have two util files).

Reply:
Moved this discussion to #120 - I think this can wait until we merged in this PR.